server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -141,6 +141,10 @@ public Task register(String type, String action, TaskAwareRequest request, boole
                 headers.put(key, httpHeader);
             }
         }
+        String user = threadContext.getTransient(Task.USER_KEY);
+        if (user != null) {
+            headers.put(Task.USER_KEY, user);
+        }
         Task task = request.createTask(taskIdGenerator.incrementAndGet(), type, action, request.getParentTask(), headers);
```

Comment:
I think because there are so many implementations of this we might should make an object `CreateTaskParams` and stuff everything in there.